packages/app/app/actions/favorites.js

```diff
@@ -18,6 +18,65 @@ export function readFavorites() {
   };
 }
 
+export function replaceFavoriteTrack(i, moveUp, length) {
```

Comment:
I don't think this and sorting should be done in the actions. It would be better to handle this functionality with a hook that creates callbacks for sorting arbitrary track lists, and use this hook in a container.